iceoryx_hoofs/include/iceoryx_hoofs/internal/concurrent/lockfree_queue/buffer.hpp

```diff
@@ -51,6 +51,7 @@ class Buffer
   private:
     using byte_t = uint8_t;
 
+    // NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays, hicpp-avoid-c-arrays)
     alignas(ElementType) byte_t m_buffer[Capacity * sizeof(ElementType)];
 
     ElementType* toPtr(index_t index) const noexcept;
```

Comment:
Please add a justification like `Required for low-level building block and wrapped in the abstraction`